client/genericclient/generic_stream_service.go

```
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2024 CloudWeGo Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package genericclient
+
+import (
+	"github.com/cloudwego/kitex/pkg/generic"
+	"github.com/cloudwego/kitex/pkg/serviceinfo"
+)
+
+func newClientStreamingServiceInfo(g generic.Generic) *serviceinfo.ServiceInfo {
+	readerWriter := g.MessageReaderWriter()
+	if readerWriter == nil {
+		// TODO: support grpc binary generic
+		return nil
```

panic may be better here (fail fast) before binary generic support.